From ee84a2a0ed040405b85913018f5be232a3744a21 Mon Sep 17 00:00:00 2001
From: Gabriel Somlo <gsomlo@gmail.com>
Date: Thu, 5 Sep 2024 20:56:02 -0400
Subject: [PATCH 2/3] fedora yosys-mancfginc patch

Fedora-specific patch:
When invoking yosys-config for examples in "make docs", need to use
relative path for includes, as they're not installed in build host
filesystem.
---
 docs/source/code_examples/extensions/Makefile  | 2 +-
 docs/source/code_examples/extensions/my_cmd.cc | 4 ++--
 docs/source/code_examples/stubnets/Makefile    | 2 +-
 docs/source/code_examples/stubnets/stubnets.cc | 4 ++--
 4 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/docs/source/code_examples/extensions/Makefile b/docs/source/code_examples/extensions/Makefile
index 74b547a20..91e1e7b28 100644
--- a/docs/source/code_examples/extensions/Makefile
+++ b/docs/source/code_examples/extensions/Makefile
@@ -11,7 +11,7 @@ CXXFLAGS=$(shell $(YOSYS)-config --cxxflags)
 DATDIR=$(shell $(YOSYS)-config --datdir)
 
 my_cmd.so: my_cmd.cc
-	$(YOSYS)-config --exec --cxx $(subst $(DATDIR),../../../../share,$(CXXFLAGS)) --ldflags -o my_cmd.so -shared my_cmd.cc --ldlibs
+	$(YOSYS)-config --exec --cxx $(subst $(DATDIR),../../../../share,$(CXXFLAGS)) -I../../../../ --ldflags -o my_cmd.so -shared my_cmd.cc --ldlibs
 
 test0.log: my_cmd.so
 	$(YOSYS) -QTl test0.log_new -m ./my_cmd.so -p 'my_cmd foo bar' -f verilog absval_ref.v
diff --git a/docs/source/code_examples/extensions/my_cmd.cc b/docs/source/code_examples/extensions/my_cmd.cc
index d52268b4a..6b67f10f3 100644
--- a/docs/source/code_examples/extensions/my_cmd.cc
+++ b/docs/source/code_examples/extensions/my_cmd.cc
@@ -1,5 +1,5 @@
-#include "kernel/yosys.h"
-#include "kernel/sigtools.h"
+#include <kernel/yosys.h>
+#include <kernel/sigtools.h>
 
 USING_YOSYS_NAMESPACE
 PRIVATE_NAMESPACE_BEGIN
diff --git a/docs/source/code_examples/stubnets/Makefile b/docs/source/code_examples/stubnets/Makefile
index 17f700c88..a15eb3005 100644
--- a/docs/source/code_examples/stubnets/Makefile
+++ b/docs/source/code_examples/stubnets/Makefile
@@ -11,7 +11,7 @@ test: stubnets.so
 	tail test1.log test2.log test3.log
 
 stubnets.so: stubnets.cc
-	yosys-config --exec --cxx --cxxflags --ldflags -o $@ -shared $^ --ldlibs
+	yosys-config --exec --cxx --cxxflags -I../../../../ --ldflags -o $@ -shared $^ --ldlibs
 
 .PHONY: clean
 clean:
diff --git a/docs/source/code_examples/stubnets/stubnets.cc b/docs/source/code_examples/stubnets/stubnets.cc
index 566d24b18..b09bc567c 100644
--- a/docs/source/code_examples/stubnets/stubnets.cc
+++ b/docs/source/code_examples/stubnets/stubnets.cc
@@ -5,8 +5,8 @@
 // binary, for any purpose, commercial or non-commercial, and by any
 // means.
 
-#include "kernel/yosys.h"
-#include "kernel/sigtools.h"
+#include <kernel/yosys.h>
+#include <kernel/sigtools.h>
 
 #include <string>
 #include <map>
-- 
2.52.0

